openidcpy/client.py

```diff
@@ -0,0 +1,163 @@
+import base64
+import json
+import urllib
+import urlparse
+
+import requests
+from jwcrypto import jwt, jwk
+from datetime import datetime
+
+
+class OidcClient(object):
+  def __init__(self, discovery_uri, client_id, client_secret=None):
+    self.discovery_uri = discovery_uri
+    self.client_id = client_id
+    self.client_secret = client_secret
+
+  def _discover(self):
+    if not hasattr(self, 'well_known'):
+      self.well_known = self._get_url(self.discovery_uri)
+      self.certs = {item['kid']: item for item in
+                    self._get_url(get_property('jwks_uri', self.well_known))[
+                      'keys']}
+      self.auth_uri = get_property('authorization_endpoint',
+                                   self.well_known)
+      self.token_uri = get_property('token_endpoint', self.well_known)
+      self.logout_uri = get_property('end_session_endpoint',
+                                     self.well_known)
+
+  def _get_url(self, url):
+    resp = None
+    try:
+      resp = requests.get(url, verify=False)
```

Comment:
Why not `with requests.get(url, verify=False) as resp`?

Reply:
`>>> with requests.get("http://www.google.com") as r:
...     print r.status_code
... 
200`

`zubairs-mbp:~ zubairnabi$ pip freeze | grep requests
requests==2.19.1`

Reply:
This is actually happening in superset that uses `requests==2.11.1`